tests/e2e/functional_test.go

```diff
@@ -30,12 +31,162 @@ type FunctionalSuite struct {
+	w.Expect().CheckCommitStatus()
+
+}
+
+// GitSync self healing occurs when resource does not match manifest in repo
+func (s *FunctionalSuite) TestSelfHealing() {
+
+	w := s.Given().GitSync("@testdata/gitsync.yaml").InitializeGitRepo("basic-resources/initial-commit").
+		When().
+		CreateGitSyncAndWait()
+	defer w.DeleteGitSyncAndWait()
+
+	// verify that test deployment is created with {replicas: 3} in spec
+	w.Expect().ResourcesExist("apps/v1", "deployments", []string{"test-deploy"})
+	w.Expect().CheckCommitStatus()
+
+	w.Expect().VerifyResourceState("apps/v1", "deployments", "test-deploy", "spec", "replicas", 3)
+
+	// apply patch to resource
+	w.ModifyResource("apps/v1", "deployments", "test-deploy", `{"spec":{"replicas":4}}`).Wait(30 * time.Second)
+
+	// verify that resource has been "healed" by resetting replica count to 3
+	w.Expect().VerifyResourceState("apps/v1", "deployments", "test-deploy", "spec", "replicas", 3)
+
+}
+
+// test behavior when changing the repoUrl of a GitSync
+func (s *FunctionalSuite) TestChangeRepoUrl() {
+
+	w := s.Given().GitSync("@testdata/gitsync.yaml").InitializeGitRepo("basic-resources/initial-commit").
+		When().
+		CreateGitSyncAndWait()
+	defer w.DeleteGitSyncAndWait()
+
+	// first repo's manifests should be applied
+	w.Expect().ResourcesExist("apps/v1", "deployments", []string{"test-deploy"})
+	w.Expect().ResourcesExist("v1", "configmaps", []string{"test-config"})
+	w.Expect().ResourcesExist("v1", "secrets", []string{"test-secret"})
+	w.Expect().CheckCommitStatus()
+
+	// changing repoUrl to point to repo2.git
+	w = s.Given().GitSync("@testdata/revised-gitsync.yaml").InitializeGitRepo("basic-resources/second-repo").
+		When().
+		UpdateGitSyncAndWait()
+
+	// verify that resources in new repository are created
+	w.Expect().ResourcesExist("apps/v1", "deployments", []string{"repo-two-deploy"})
+	w.Expect().ResourcesExist("v1", "configmaps", []string{"repo-two-config"})
+	w.Expect().ResourcesExist("v1", "secrets", []string{"repo-two-secret"})
+	w.Expect().CheckCommitStatus()
+
+	// old resources are deleted as their manifests do not exist in new repository
+	w.Expect().ResourcesDontExist("apps/v1", "deployments", []string{"test-deploy"})
+	w.Expect().ResourcesDontExist("v1", "configmaps", []string{"test-config"})
+	w.Expect().ResourcesDontExist("v1", "secrets", []string{"test-secret"})
+
+}
+
+func (s *FunctionalSuite) TestKustomize() {
+
+	w := s.Given().GitSync("@testdata/kustomize-gitsync.yaml").InitializeGitRepo("kustomize/initial-commit").
+		When().
+		CreateGitSyncAndWait()
+	defer w.DeleteGitSyncAndWait()
+
+	w.Expect().ResourcesExist("apps/v1", "deployments", []string{"test-deploy"})
+	w.Expect().ResourcesExist("v1", "configmaps", []string{"test-config"})
+	w.Expect().ResourcesExist("v1", "secrets", []string{"test-secret"})
+	w.Expect().CheckCommitStatus()
+
+	// w.PushToGitRepo("kustomize/initial-commit", []string{"deployment.yaml"}, true)
```

Comment:
This is commented out as I was not seeing the deployment getting deleted when syncing with the commit.